src/amo/components/MoreAddonsByAuthorsCard/index.js

```diff
@@ -0,0 +1,203 @@
+/* @flow */
+import makeClassName from 'classnames';
+import deepEqual from 'deep-eql';
+import * as React from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+
+import AddonsCard from 'amo/components/AddonsCard';
+import {
+  fetchAddonsByAuthors,
+  getAddonsForUsernames,
+  getLoadingForAuthorNames,
+} from 'amo/reducers/addonsByAuthors';
+import {
+  ADDON_TYPE_DICT,
+  ADDON_TYPE_EXTENSION,
+  ADDON_TYPE_LANG,
+  ADDON_TYPE_THEME,
+} from 'core/constants';
+import { withErrorHandler } from 'core/errorHandler';
+import translate from 'core/i18n/translate';
+import type { AddonsByAuthors } from 'amo/reducers/addonsByAuthors';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
+import type { DispatchFunc } from 'core/types/redux';
+
+import './styles.scss';
+
+
+type Props = {|
+  addons?: Array<AddonType> | null,
+  addonType?: string,
+  authorNames: Array<string>,
+  className?: string,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  forAddonSlug?: string,
+  i18n: I18nType,
+  loading?: boolean,
+  numberOfAddons: number,
+
+  // AddonCards prop this component also accepts
+  showSummary?: boolean,
+  type?: 'horizontal',
+|};
+
+export class MoreAddonsByAuthorsCardBase extends React.Component<Props> {
+  static defaultProps = {
+    showSummary: false,
+    type: 'horizontal',
+  }
+
+  componentWillMount() {
+    const { addons, addonType, authorNames, forAddonSlug } = this.props;
+
+    if (!addons) {
+      this.dispatchFetchAddonsByAuthors({ addonType, authorNames, forAddonSlug });
+    }
+  }
+
+  componentWillReceiveProps({
+    addonType: newAddonType,
+    authorNames: newAuthorNames,
+    forAddonSlug: newForAddonSlug,
+  }: Props) {
+    const {
+      addonType: oldAddonType,
+      authorNames: oldAuthorNames,
+      forAddonSlug: oldForAddonSlug,
+    } = this.props;
+
+    if (
+      oldAddonType !== newAddonType ||
+      oldForAddonSlug !== newForAddonSlug ||
+      !deepEqual(oldAuthorNames, newAuthorNames)
+    ) {
+      this.dispatchFetchAddonsByAuthors({
+        addonType: newAddonType,
+        authorNames: newAuthorNames,
+        forAddonSlug: newForAddonSlug,
+      });
+    }
+  }
+
+  dispatchFetchAddonsByAuthors({ addonType, authorNames, forAddonSlug }: Object) {
+    this.props.dispatch(fetchAddonsByAuthors({
+      addonType,
+      authors: authorNames,
```

Comment:
This seems to be the only place where `authors` is the name of the parameter as opposed to `authorNames`. For consistency it might make sense to rename the parameter in `fetchAddonsByAuthors` in the reducer and then you wouldn't need to map `authorNames` to `authors`. I'll add a comment in the reducer as well.

Reply:
Agreed; I didn't change this because it was old code and there were other places I didn't want to change at first... but yeah this should be tidied up 👍